tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+import java.util.logging.Logger;
+
+/**
+ *     The LRA model guarantees eventual consistency but does not say when
+ *     participants will be brought into a consistent state. When a request
+ *     to close/cancel an LRA is received, the implementation is allowed to
+ *     delay sending out the complete/compensate callbacks (the specification
+ *     only requires that an implementation must eventually send them).
+ *
+ *     This SPI provides a mechanism for implementations to indicate when
+ *     it knows that callbacks have been sent and responses received or when
+ *     callbacks have been issued and all participants have reached an end state.
+ *
+ */
+public interface LRARecoveryService {
+
+    Logger LOGGER = Logger.getLogger(LRARecoveryService.class.getName());
```

Comment:
Would it be possible to move this into the default implementation since an implementation may not need to log (eg it could use callbacks instead of polling) or may use a different logging framework.

Reply:
The implementation can always implement this method. It only makes sense, because this is expected behavior. I can remove logging or just replace it with System.out.println?

Reply:
meant to move the `getLogger` call into the default method so it would like:

``` java
    default void waitForRecovery(URI lraId) throws LRACallbackException {
         Logger LOGGER = Logger.getLogger(LRARecoveryService.class.getName());
         int counter = 0;

        do {
            LOGGER.info("Recovery attempt #" + ++counter);
        } while (!waitForEndPhaseReplay(lraId));
        LOGGER.info("LRA " + lraId + "has finished the recovery");
    }
```